modules/trusted-firmware-m/Kconfig

```diff
@@ -0,0 +1,140 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+config TFM_MINIMAL
+	bool "Use minimal TFM build"
+	imply TFM_IPC
+
+if !TFM_MINIMAL
+source "$(ZEPHYR_BASE)/modules/trusted-firmware-m/Kconfig"
+
+endif
+
+if TFM_MINIMAL
+
+config TFM_MCUBOOT_IMAGE_NUMBER
+	int
+	default 1
+
+# The HASH module is needed to provide RNG functionality
+config TFM_CRYPTO_HASH_MODULE_ENABLED
+	bool
+	default y
+
+config TFM_PARTITION_CRYPTO
+	bool
+	default y
+
+config TFM_PARTITION_PLATFORM
+	bool
+	default y
+
+source "$(ZEPHYR_BASE)/modules/trusted-firmware-m/Kconfig.tfm"
+
+config MBEDTLS_SHA1_C
+	default n
+
+config MBEDTLS_SHA512_C
+	default n
+
+# SHA256 is enabled since we don't override its default
+
+config MBEDTLS_TLS_LIBRARY
+	default n
+
+config MBEDTLS_X509_LIBRARY
+	default n
+
+config MBEDTLS_ENABLE_HEAP
+	default n
+
+config MBEDTLS_PK_WRITE_C
+	default n
+
+config MBEDTLS_RSA_C
+	default n
+
+config MBEDTLS_DHM_C
+	default n
+
+config MBEDTLS_PK_C
+	default n
+
+config MBEDTLS_ECP_C
+	default n
+
+config MBEDTLS_HMAC_DRBG_C
+	default n
+
+config MBEDTLS_CIPHER_MODE_CBC
+	default n
+
+config MBEDTLS_CIPHER_MODE_CTR
+	default n
+
+config MBEDTLS_CMAC_C
+	default n
+
+config MBEDTLS_CCM_C
+	default n
+
+config MBEDTLS_GCM_C
+	default n
+
+config MBEDTLS_CHACHA20_C
+	default n
+
+config MBEDTLS_PK_C
+	default n
+
+config MBEDTLS_POLY1305_C
+	default n
+
+config MBEDTLS_PSA_CRYPTO_STORAGE_C
```

Comment:
Here and above, this is a long list with `default n`.
Are all those `default y` in existing code ?

Reply:
They were all being implied in an earlier version of the TFM-integration branch. I have rebased on top of latest now, will see if I can remove some of these

Reply:
Removing these 
![image](https://user-images.githubusercontent.com/1547348/121012165-bb390f00-c797-11eb-92c4-07497726bd12.png)

Reply:
The rest are default y

Reply:
thanks for checking that up.

We could consider to place the mbedtls config in a `Kconfig.mbedtls.defconfig` and then source that file conditionally.
That could make it clearer that this is purely defconfig settings.

Reply:
Good point, I will do that, since it makes the intent of this kconfig clearer.